### PR TITLE
Foxtrot dashboards for handlers

### PR DESCRIPTION
**Pull request by sebastian-bojara**
https://app.asana.com/0/1203669339406776/1202458568143107/f

Screenshot 2023-04-25 at 10 40 06

https://github.com/Accedian/Foxtrot/pull/37

### CODE REVIEW

**Comment by robsztal** [comment]
1. WS server messages duration - do we need all quantiles ? check alert service for that
2. WS server messages - should be only 200 visible and separate counter for not 200's
3. Postgres call duration - same as 1

**Comment by robsztal** [comment]
IMO ws server messages should be separated as we have in alert service and represented as counter

File: dashboards/foxtrot_service_dashboard.json [hunk 1]
```diff
@@ -120,7 +120,208 @@
...
+            "expr": "foxtrot_ws_server_handle_duration_count",
+            "format": "time_series",
+            "instant": false,
+            "legendFormat": "Message type \"{{message_type}}\" resulted in related HTTP code {{related_http_code}}",
+            "range": true,
+            "refId": "A"
+          }
+        ],
+        "title": "WS Server messages",
+        "type": "timeseries"
```
Comment (robsztal) [inline dashboards/foxtrot_service_dashboard.json:219]: I am not sure if timeseries will be working on gcr graphana

File: dashboards/foxtrot_service_dashboard.json [hunk 2]
```diff
@@ -120,7 +120,208 @@
         "type": "timeseries"
       },
       {
-        "datasource": "Prometheus",
+        "datasource": {
+          "type": "prometheus",
+          "uid": "PBFA97CFB590B2093"
```
Comment (robsztal) [inline dashboards/foxtrot_service_dashboard.json:125]: this will probably not work. should be without uid

"datasource": "Prometheus",

**Comment by robsztal** [comment]
that datasource needs to be changed for sure and some other thing because it is not working now on jw

image

File: dashboards/foxtrot_service_dashboard.json (hunk 2 above)
Comment (robsztal) [reply dashboards/foxtrot_service_dashboard.json:125]: this applies to all fields in json file

**Comment by sebastian-bojara** [comment]
@robsztal 
> WS server messages - should be only 200 visible and separate counter for not 200's

Why only? And why would we need another counter if we can see it from legend?

**Comment by sebastian-bojara** [comment]
> IMO ws server messages should be separated as we have in alert service and represented as counter

Why? What does it say, how it can help? From time series we can read amounts, amount per message type. We can correlate it (e.g. with deployments).

File: dashboards/foxtrot_service_dashboard.json (hunk 2 above)
Comment (sebastian-bojara) [reply dashboards/foxtrot_service_dashboard.json:125]: Thank you!

File: dashboards/foxtrot_service_dashboard.json (hunk 1 above)
Comment (sebastian-bojara) [reply dashboards/foxtrot_service_dashboard.json:210]: Works locally, why not?

File: dashboards/foxtrot_service_dashboard.json [hunk 3]
```diff
@@ -120,7 +120,208 @@
...
+        },
+        "targets": [
+          {
+            "datasource": {
+              "type": "prometheus",
+              "uid": "PBFA97CFB590B2093"
+            },
+            "editorMode": "builder",
+            "exemplar": false,
+            "expr": "foxtrot_ws_server_handle_duration_count",
```
Comment (mrdunski) [inline dashboards/foxtrot_service_dashboard.json:201]: legendFormat is not in line with expr. Do you mean to use aggregation function, like `sum(...) by (message_type, related_http_code)`?

Comment (mrdunski) [reply dashboards/foxtrot_service_dashboard.json:201]: Also, is it useful to duration count without any kind of rate?

File: dashboards/foxtrot_service_dashboard.json (hunk 1 above)
Comment (robsztal) [reply dashboards/foxtrot_service_dashboard.json:210]: When I was creating dashboard I had a problem w timeseries (something about plugin not supported) but I have checked and on jwatson there is already timeseries so we can close that topic

File: dashboards/foxtrot_service_dashboard.json (hunk 3 above)
Comment (mrdunski) [reply dashboards/foxtrot_service_dashboard.json:201]: (the same thing goes to other expressions like this one)

**Comment by robsztal** [comment]
> @robsztal
> 
> > WS server messages - should be only 200 visible and separate counter for not 200's
> 
> Why only? And why would we need another counter if we can see it from legend?

Its just my opinion but I think it will be easier to analyze - we have the same thing in alert service -> good processed msgs and errors on the other side but if you think one graph is enough go with it :)

**Comment by sebastian-bojara** [comment]
@robsztal There was the simple counter. I looked at it and thought it was fine and can be better. Time series can answer more questions: which message type produces an error? what kind of error is it? when an error happened?

**Comment by sebastian-bojara** [comment]
grafana

**Comment by sebastian-bojara** [comment]
https://github.com/Accedian/Foxtrot/pull/41

**Comment by sebastian-bojara** [comment]
@robsztal 
I don't ping Patryk as he is really busy right now fixing the bug.

**Pull request merged**